blog/_posts/2021-07-06-distributed-rdataframe.md

```diff
@@ -0,0 +1,105 @@
+their applications from within the same interface while steering the computations to an Apache Spark cluster.
+
+## In detail
+
+[RDataFrame](https://root.cern/doc/master/classROOT_1_1RDataFrame.html) is ROOT's high-level interface for data
+analysis and since its release in 6.14 has seen tons of usages in real world analysis use cases and other examples in
+the wild. Parallelism has always been a staple of its design with support for executing the event loop on all cores
+of a machine thanks to implicit multi-threading. Since ROOT 6.24, this aspect of RDataFrame has been enhanced further
+with distributed computing capabilities, allowing users to run their analysis on multi-node computing clusters through
+widely used frameworks such as [Apache Spark](https://spark.apache.org/), [Dask](https://dask.org/) or
+[AWS Lambda](https://aws.amazon.com/lambda/). Here is an example usage of the tool (requires the Python `pyspark`
+package):
+
+```python
+import ROOT
+ 
+# Point RDataFrame calls to the Spark specific RDataFrame
+RDataFrame = ROOT.RDF.Experimental.Distributed.Spark.RDataFrame
+ 
+# It still accepts the same constructor arguments as traditional RDataFrame
+df = RDataFrame("mytree", "myfile.root")
+ 
+# Continue the application with the traditional RDataFrame API
+sum = df.Filter("x > 10").Sum("y")
+h = df.Histo1D("x")
+ 
+print(sum.GetValue())
+h.Draw()
+```
+
+The main goal is to support distributed execution of any RDataFrame application. This has led to the creation of a 
+**Python** package that connects the RDataFrame API (available in Python through PyROOT) and the APIs of
+distributed computing frameworks, which are offered in Python in the vast majority of cases. Another key goal is to be
+able to support as many backends as possible, in order to avoid being tied to any particular infrastructure. This is
+achieved through a modular implementation that defines a generic task (representing the RDataFrame computation graph) to
+be executed on data. The input dataset is logically split into many ranges of entries, which will be sent to the
+distributed nodes for processing. Each range will then be paired to the generic task and submitted to the computing
+framework via a specific backend implementation. An added benefit of using RDataFrame is that the distributed
+tasks run C++ computations. This is made possible by PyROOT and cling. Currently the package offers support for running
+the application on a Spark cluster, but the package design will make it possible to add many more backends in the future.
+For example, backends for Dask and AWS Lambda have already been implemented and demonstrated at different conferences
+[[1]](#dasksummit)[[2]](#hips21). They will be made available in future ROOT releases.
+
+Great attention is being put towards ensuring that running the RDataFrame computation graph scales well across
+multiple computing nodes and different backend implementations. This has been shown since the first stages of the
```

Comment:
```suggestion
Excellent scaling is paramount for this distributed RDataFrame implementation, to ensure you can run the RDataFrame computation graph scales efficiently across
multiple computing nodes and different backend implementations. This has been shown since the first stages of the
```